pandas/core/internals.py

```diff
@@ -2559,7 +2540,7 @@ def _try_coerce_result(self, result):
 
     @property
     def _box_func(self):
-        return tslib.Timestamp
+        return lambda x: tslib.Timestamp(x, freq=self.freq, tz=self.tz)
```

Comment:
you can just return
```self.values._box_func()``, no need to add any attributes.

Reply:
That would work for DatetimeTZBlock, but not for DatetimeBlock.  This edit is to a) let them share the implementation and b) make the implementation match that of DatetimeIndex to move in the direction of sharing implementations.